kitsune/questions/migrations/0022_remove_aaqconfig_pinned_articles_and_more.py

```diff
@@ -0,0 +1,86 @@
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+def migrate_pinned_articles(apps, schema_editor):
+    """
+    Migrate the pinned articles associated with AAQConfig objects to references
+    to PinnedArticleConfig objects that have been populated with those same
+    pinned articles.
+    """
+    AAQConfig = apps.get_model("questions", "AAQConfig")
+    PinnedArticleConfig = apps.get_model("wiki", "PinnedArticleConfig")
+
+    already_created: dict[frozenset, PinnedArticleConfig] = {}
+
+    for aaq_config in AAQConfig.objects.filter(pinned_articles__isnull=False).prefetch_related(
+        "pinned_articles"
+    ):
+        # Create a unique and hashable signature for this set of pinned articles.
+        pinned_articles = aaq_config.pinned_articles.all()
+        signature = frozenset(article.id for article in pinned_articles)
+
+        # If this same set of pinned articles hasn't already been created, create it.
+        if signature not in already_created:
+            title = f"Auto-migrated AAQ Set {len(already_created) + 1}"
+            pinned_article_config = PinnedArticleConfig.objects.create(title=title)
+            pinned_article_config.pinned_articles.set(pinned_articles)
+            already_created[signature] = pinned_article_config
+            print(
+                f'Created PinnedArticleConfig "{title}" with {len(signature)} pinned article(s).'
+            )
+
+        aaq_config.pinned_article_config = already_created[signature]
+        aaq_config.save()
+        print(
+            f'Associated AAQConfig "{aaq_config.title}" with PinnedArticleConfig '
+            f'"{aaq_config.pinned_article_config.title}".'
+        )
+
+
+def reverse_migrate_pinned_articles(apps, schema_editor):
```

Comment:
Let's remove the reverse. If this is run with the `pinned_articles` dropped from the schema it's going to fail and will block all the backwards operations under the `questions` app.